docker/Dockerfile.indy

```diff
@@ -243,26 +242,22 @@ LABEL summary="$SUMMARY" \
     io.k8s.description="$DESCRIPTION" \
     io.k8s.display-name="aries-cloudagent $acapy_version" \
     name="aries-cloudagent" \
-    version="$acapy_version" \
+    acapy.version="$acapy_version" \
     maintainer=""
 
-# Create standard directories to allow volume mounting and set permissions
-# Note: PIP_NO_CACHE_DIR environment variable should be cleared to allow caching
-RUN mkdir -p $HOME/.aries_cloudagent
-
-# The root group needs access the directories under $HOME/.indy_client for the container to function in OpenShift.
-# Also ensure the permissions on the python 'site-packages' folder are set correctly.
-RUN chmod -R ug+rw $HOME/.aries_cloudagent
-
+# Install ACA-py from the wheel as $user,
+# and ensure the permissions on the python 'site-packages' folder are set correctly.
 COPY --from=acapy-builder /src/dist/aries_cloudagent*.whl .
-
-# Install ACA-py from the wheel.
 RUN aries_cloudagent_package=$(find ./ -name "aries_cloudagent*.whl" | head -n 1) && \
     echo "Installing ${aries_cloudagent_package} ..." && \
     pip install --no-cache-dir --find-links=. ${aries_cloudagent_package}${acapy_reqs} && \
-    rm aries_cloudagent*.whl
+    rm aries_cloudagent*.whl && \
+    chmod +rx $(python -m site --user-site)
 
 # Clean-up unneccessary build dependencies and reduce final image size
-# RUN apt-get purge -y --auto-remove build-essential
+USER root
+RUN apt-get purge -y --auto-remove build-essential
+
+USER $user
```

Comment:
This being right at the end of the dockerfile stands out to me as not being ideal. Not that it's perfectly optimized anyways but these lines will basically always run. Is it possible to rearrange or produce another stage that can omit the build essentials from the start?

Reply:
Depending on the availability of binary packages for the given platform `build-essential` is required during the `aca-py` install to build and install some of the dependencies.  It can only be removed after the install is complete.

Reply:
@dbluhm, does that make sense?

Reply:
Ah, I see. I retain an itch that there might be a better way but I don't think it's worth holding this PR up; sorry for the slow response.